encode.avs

```diff
@@ -156,35 +156,41 @@ if (prescaleFactor > 1 && !hd) {
 #	Aspect ratio correction for SD encodes
 #	if dimensions are not multiples of 4, codecs freak out, so we enforce mod 4
 mod		= 4
-hdHeight= 2160
 height	= last.height.ForceModulo(mod, true)
 width	= (ARCorrection \
 		? height * wAspect / hAspect \
 		: last.width) \
 		.ForceModulo(mod, true)
 
-#	Remember SD frame size for subYpos HD tweaks
-wARC	= width
-hARC	= height
-
 #	Actually go HD if we need
 if (hd) {
-	height = hdHeight
-}
-
-width	= (ARCorrection \
-		? height * wAspect / hAspect \
-		: height * last.width / last.height) \
-		.ForceModulo(mod, true)
-hStretch= height / last.height
-
-# YouTube has a 4K width limit of 3840
-if (hd && (width > 3840)) {
-	width = 3840
-	height = (ARCorrection \
-		? width * hAspect / wAspect \
-		: width * last.height / last.width) \
-		.ForceModulo(mod, true)
+	vertical        = ((width < height) && !ARCorrection) \
+		|| (ARCorrection && ((float(wAspect) / hAspect) < 1))

```

Comment:
Since `width` and `height` are already set right above for SD, can't we simply rely on them here without the extra checks?

Reply:
I didn't realize that. I was focused too much on making sure vertical aspect ratio correction with values like 3:4 didn't break anything.